### PR TITLE
Fix(bigquery)!: parse information schema views into a single identifier

### PR DESCRIPTION
**Pull request by georgesittas**
Behavior in main today:

```python
>>> from sqlglot import parse_one
>>> parse_one("select * from db.information_schema.tables", "bigquery")
Select(
  expressions=[
    Star()],
  from=From(
    this=Table(
      this=Identifier(this=tables, quoted=False),
      db=Identifier(this=information_schema, quoted=False),
      catalog=Identifier(this=db, quoted=False))))
>>> parse_one("select * from c.db.information_schema.tables", "bigquery")
Select(
  expressions=[
    Star()],
  from=From(
    this=Table(
      this=Dot(
        this=Identifier(this=information_schema, quoted=False),
        expression=Identifier(this=tables, quoted=False)),
      db=Identifier(this=db, quoted=False),
      catalog=Identifier(this=c, quoted=False))))
```

This PR makes it so that we will always represent `INFORMATION_SCHEMA.X` views using a single `Identifier` expression for BigQuery. The motivation here is that, previously, we wouldn't properly qualify information schema references, because we'd think there's a db/catalog present.

Choosing `Identifier` over a `Dot` is motivated by the fact that we the schema module can't handle multiple mapping depths, so this should keep the nesting consistent even if there are information schema view references.

Xref: https://github.com/TobikoData/sqlmesh/issues/3317

### CODE REVIEW

**Review by barakalon — COMMENTED** [review]
I’m mostly familiar with MySQL’s information_schema. In MySQL, information_schema _is_ a database, so I think this would be wrong?

Does this PR imply that Bigquery creates default information_schema tables in each database? And if so, I wonder how common that is.

**Comment by georgesittas** [comment]
> I’m mostly familiar with MySQL’s information_schema. In MySQL, information_schema _is_ a database, so I think this would be wrong?
> 
> Does this PR imply that Bigquery creates default information_schema tables in each database? And if so, I wonder how common that is.

Ah, got it - thanks for catching this. I did assume there would be an information schema under each db, but I'll double-check this assumption.

Regarding BigQuery, yes, they [need](https://cloud.google.com/bigquery/docs/information-schema-intro#syntax) the information schema to be qualified with a region or dataset:

> An INFORMATION_SCHEMA view needs to be qualified with a dataset or region.

**Comment by georgesittas** [comment]
@barakalon it turns out your hunch was right, none of the other engines treat the information schema views like BigQuery does, so this fix really needs to be specific to it. Thanks for catching this, should've investigated more thoroughly.

**Comment by georgesittas** [comment]
There's some work needing to be done here around 4-part table references in the schema module, because right now we're expecting the references to have the same depth but information schema views break this invariant.

**Review by barakalon — APPROVED** [review]

File: tests/test_optimizer.py [hunk 1]
```diff
@@ -315,7 +315,7 @@ def test_qualify_columns(self, logger):
                 ),
                 dialect="bigquery",
             ).sql(),
-            'WITH "x" AS (SELECT "y"."a" AS "a" FROM "DB"."y" AS "y" CROSS JOIN "a"."b"."INFORMATION_SCHEMA"."COLUMNS" AS "COLUMNS") SELECT "x"."a" AS "a" FROM "x" AS "x"',
+            'WITH "x" AS (SELECT "y"."a" AS "a" FROM "DB"."y" AS "y" CROSS JOIN "a"."b"."INFORMATION_SCHEMA.COLUMNS" AS "columns") SELECT "x"."a" AS "a" FROM "x" AS "x"',
```
Comment (georgesittas) [inline tests/test_optimizer.py:318]: This change should be safe because BigQuery aliases are case-insensitive.

**Review by barakalon — APPROVED** [review]

**Pull request merged**